### ISSUE TITLE
Tensorflow Numpy (TF-Numpy) as backend (or as Replacement !)

### ISSUE THREAD

**Issue by SaitejaUtpala** [geomstats/geomstats#1037]
TF-Numpy is Tensorflow implementation of Numpy API. It is in experimental stage. https://www.tensorflow.org/guide/tf_numpy

  Here are key things
-   we can get huge speedups for any numpy code  (all we have to change is imports).
-   currently all numpy operations are supported. But  like tensorflow tensors,  TF-Numpy arrays are immutable. But their plan is to change that move closer to numpy-style (i.e., you can use assignment operations on them  )
-  we can differentiate through numpy code 
-  accelerate numpy code on GPUS/TPUS

There are two possibilities 

- since `autograd` is  no longer actively developed and using `jax` is turning out to be difficult(#922 ) . we could use tensorflow numpy as ready-to-use replacement for numpy. 
- keep original numpy as backend and create a new backend called TF-numpy

**Comment by nguigs** [comment]
Seems definitely worth testing it! Does it work in float or double by default? The major advantage of our current numpy backend is numerical precision in double. I think We should keep a pure numpy (even without autograd) and add a new backend with tf-numpy.
There might be a bit of work factoring the autograd module, as in #984

**Comment by SaitejaUtpala** [comment]
> Seems definitely worth testing it! Does it work in float or double by default? The major advantage of our current numpy backend is numerical precision in double. I think We should keep a pure numpy (even without autograd) and add a new backend with tf-numpy.
> There might be a bit of work factoring the autograd module, as in #984

Hey @nguigs .
- I think we can set default type to what we desire . ex. In pytorch this is done as `torch.set_default_tensor_type(torch.DoubleTensor)`  for `float64`. So I am not sure what is a major advantage for numpy backend? could you elaborate on this?
- yup. I tested, it is indeed float64 
```
import numpy as np
print(np.array(1.).dtype) # this would print float64

import tensorflow.experimental.numpy as tnp
tnp.experimental_enable_numpy_behavior()
print(tnp.array(1.).dtype) # this would be <dtype: 'float64'>
```
- Here is another thing.  tnp tensors can be passed to numpy functions and numpy arrays can be passed to tnp functions (It is not advised to do so for performance related reasons). Also, as one would expect tnp tensors is compatible with tensorflow tensors/functions
